cmd/gitlab.go

```diff
@@ -466,40 +469,39 @@ func (c *Client) findProjectsFromWildcards() error {
 	return nil
 }
 
-func (c *Client) pollProjectsWith(numWorkers int, until <-chan struct{}, projects ...Project) []error {
-	var errs []error
+func (c *Client) pollProjectsWith(numWorkers int, doing func(Project) error, until <-chan struct{}, on ...Project) <-chan error {
 	errorStream := make(chan error)
-	defer close(errorStream)
-	projectsToPoll := make(chan Project, len(projects))
+	projectsToPoll := make(chan Project, len(on))
+	// sync closing the error channel via a waitGroup
+	wg := sync.WaitGroup{}
+	wg.Add(numWorkers)
+
 	// spawn maximum_projects_poller_workers to process project polling in parallel
 	for w := 0; w < numWorkers; w++ {
-		go func() {
-			for {
+		go func(wg *sync.WaitGroup) {
+			defer wg.Done()
+			for p := range projectsToPoll {
 				select {
 				case <-until:
 					return
-				case p := <-projectsToPoll:
-					if e := c.pollProject(p); e != nil {
-						errorStream <- e
-					}
+				case errorStream <- doing(p):
 				}
 			}
-		}()
+		}(&wg)
```

Comment:
```suggestion
		}(wg)
```

Reply:
Tied to [above](https://github.com/mvisonneau/gitlab-ci-pipelines-exporter/pull/91#discussion_r411925141) we really need to pass a pointer here or the `wg.Done()` will have a wrong behaviour.
So if you don't like passing the address in the closure, we can implement the suggested code and initialise the variable as `wg := &sync.WaitGroup{}`.